### PR TITLE
added error messages and memory reset

### PR DESCRIPTION
**Pull request by johncalesp**
- Added external method to clear cache in GPU memory
- Added error codes for habitat and energy in protobuf
- Modified try-except blocks for habitat and energy

### CODE REVIEW

File: deepview_profile/functionality_utils.py
Comment (michaelshin) [inline deepview_profile/functionality_utils.py:1]: I'd call this file memory_utils or something similar

File: deepview_profile/analysis/session.py [hunk 1]
```diff
@@ -150,47 +150,56 @@ def energy_compute(self) -> pm.EnergyResponse:
             for _ in range(iterations):
                 iteration(*inputs)
             energy_measurer.end_measurement()
+            resp.total_consumption = energy_measurer.total_energy()/float(iterations)
```
Comment (michaelshin) [inline deepview_profile/analysis/session.py:153]: Does this change mean a user has to enable CPU monitoring to get energy measurements? What if a user doesn't have access like in cloud instances?

File: deepview_profile/analysis/session.py [hunk 2]
```diff
@@ -210,102 +219,115 @@ def habitat_compute_threshold(self, runnable, context):
...
+                pred.runtime_ms = predicted_trace.run_time_ms
+                resp.predictions.append(pred)
+
+            print(f"returning {len(resp.predictions)} predictions.")
+        except AnalysisError as ex:
+            message = str(ex)
+            logger.error(message)
+            resp.analysis_error.error_message = message
+        except:
+            logger.error("There was an error running deepview predict")
```
Comment (michaelshin) [inline deepview_profile/analysis/session.py]: Change "deepview predict" to "DeepView Predict"

**Review by michaelshin — CHANGES_REQUESTED** [review]

File: deepview_profile/analysis/session.py (hunk 1 above)
Comment (johncalesp) [reply deepview_profile/analysis/session.py:153]: Actually no, the code is the same as before. I tried on my PC removing access to pyRAPL and the profile will only measure GPU power consumption.

File: deepview_profile/analysis/session.py (hunk 2 above)
Comment (johncalesp) [reply deepview_profile/analysis/session.py]: got it

File: deepview_profile/functionality_utils.py
Comment (johncalesp) [reply deepview_profile/functionality_utils.py:1]: renamed to utils.py

File: deepview_profile/analysis/session.py (hunk 1 above)
Comment (michaelshin) [reply deepview_profile/analysis/session.py:153]: After looking over the code with John, we determined that the exception is already handled in the measurement

**Review by michaelshin — APPROVED** [review]

**Pull request merged**